### PR TITLE
fix(config): override config values instead of merging

### LINKED ISSUE #1863: Delete chat and others options
Hi, i've a client with an chat, this chats work properly and i don't wanna use the jitsi-meet chat.

can the api a remove this chat? in the documentation i saw a option to toggle the chat but this option only change the object's display.

I just wanna remove the chat's icon and call to number.

Greetings to everyone.

thanks for build jitsi-meet.

greetings from Venezuela.

### PR DESCRIPTION
**Pull request by virtuacoplenny**
Iterate over objects and copy over primitives and arrays
instead of using _.merge, as merge will not replace a config
entry completely. For arrays in a target object, the arrays
will have its indices replaced. This means if a source array
is empty, the target array will be left alone. Similarly,
if the target array is longer than a source array, there
will be indices not touched in the target array.

This commit fixes #1863.

### CODE REVIEW

**Comment by virtuacoplenny** [comment]
@saghul, I'm uncertain of what edge cases to test for this. Please advise what I should look out for.

**Comment by saghul** [comment]
We might need to expand this to operate in a more granular way. I'd ay that for objects (for instance the p2p section) we want merging, but for arrays we want assigning.

File: react/features/base/config/functions.js [hunk 1]
```diff
@@ -241,3 +241,32 @@ export function setConfigFromURLParams() {
...
+    keys.forEach(key => {
+        const currentValue = targetObject[key];
+        const newValue = sourceObject[key];
+
+        if (typeof currentValue !== 'undefined'
+            && typeof newValue === 'object'
+            && !Array.isArray(newValue)) {
+            _merge(currentValue, newValue);
+        } else {
+            targetObject[key] = newValue;
```
Comment (saghul) [inline react/features/base/config/functions.js]: maybe copy arrays here, instead of copying just the reference

File: react/features/base/config/functions.js [hunk 2]
```diff
@@ -241,3 +241,32 @@ export function setConfigFromURLParams() {
...
+ * overriden or extended.
+ * @param {Object} sourceObject - The object whose values will be put into
+ * targetObject.
+ * @private
+ * @returns {void}
+ */
+function _merge(targetObject, sourceObject) {
+    const keys = Object.keys(sourceObject);
+
+    keys.forEach(key => {
```
Comment (saghul) [inline react/features/base/config/functions.js]: use for..in instead

File: react/features/base/config/functions.js [hunk 3]
```diff
@@ -241,3 +241,32 @@ export function setConfigFromURLParams() {
...
+        const currentValue = targetObject[key];
+        const newValue = sourceObject[key];
+
+        if (typeof currentValue !== 'undefined'
```
Comment (saghul) [inline react/features/base/config/functions.js]: please leave a comment explaining why you filter arrays.

**Comment by virtuacoplenny** [comment]
Wrote some rudimentary tests here: https://jsbin.com/rikameyaze/1/edit?js,console.

File: react/features/base/config/functions.js [hunk 4]
```diff
@@ -177,7 +177,7 @@ export function overrideConfigJSON(
                 logger.info(
                     `Extending ${configName} `
                     + `with: ${JSON.stringify(configJSON)}`);
-                _.merge(configObj, configJSON);
+                _merge(configObj, configJSON);
```
Comment (lyubomir) [inline react/features/base/config/functions.js]: Cannot we merely use `_.mergeWith` and `objValue => Array.isArray(objValue) ? objValue : undefined` as its `customizer`?

**Comment by alfredorods** [comment]
Hey, thanks for fix my issue.

These change are available on meet.jit.si? if this change are available i wanna test my application.

Sorry for my english.

**Comment by virtuacoplenny** [comment]
No, the changes are not on meet.jit.si. I'm unsure when the next deploy to meet.jit.si will be.
